Update doc strings to make description of is_causal consistent for nn.Transformer and nn.MHA

mikekgfb opened the pull request:
Summary: Update doc strings to make description of is_causal consistent for nn.Transformer and nn.MHA

Test Plan: sandcastle & github CI/CD

Differential Revision: D45737197

mikaylagawarecki commented on torch/nn/modules/transformer.py:
```diff
@@ -361,9 +365,20 @@ def forward(self, tgt: Tensor, memory: Tensor, tgt_mask: Optional[Tensor] = None
+                incorrect execution, including forward and backward
+                compatibility.
+            memory_is_causal: If specified, applies a causal mask as
+                memory mask.
```
nit: perhaps would be nice to format like the argument name
```suggestion
                ``memory_mask``.
```

here and similarly for the other variants (i.e. If specified, applies a causal mask as ``<formatted argname>``)

mikekgfb replied:
Good suggestion! Done.

mikekgfb commented:
@pytorchbot merge

Commit:
Refactor is_causal w/o moduleinfo (#101089)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/101089

Update doc strings for consistency in wording that is_causal is a hit for nn.Transformer, nn.MHA

Test Plan: sandcastle & github

Reviewed By: mikaylagawarecki

Differential Revision: D45737197

fbshipit-source-id: 6d120274bfd408e795101e91b62f06aef23976b4

mikekgfb commented:
@pytorchbot merge
